### PR TITLE
fix(work-item-widget): fix component updating when navigating differe…

### PR DESCRIPTION
**Pull request by aptmac**
…nt spaces

This PR addresses issue #3946 [[0]](https://github.com/openshiftio/openshift.io/issues/3946), in which the dashboard work-items-widget does not update when changing. This was because the component wasn't calling the work-items service when the context is changed, and even if it was, the variables containing the counts weren't being reset.

Here are a couple of gifs showing the before/after behaviour of this PR.

**Before:**
before

**After:**
after

[0] https://github.com/openshiftio/openshift.io/issues/3946

### CODE REVIEW

**Comment by alien-ike** [comment]
### Ike Plugins (test-keeper)

Thank you @aptmac for this contribution!



It appears that no tests have been added or updated in this PR.

Automated tests give us confidence in shipping reliable software. Please add some as part of this change.

If you are an admin or the reviewer of this PR and you are sure that no test is needed then you can use the command `/ok-without-tests` as a comment to make the status green.


For more information please head over to official [documentation](http://arquillian.org/ike-prow-plugins/#_test_keeper_plugin). You can find there how to configure the plugin.

**Comment by aptmac** [comment]
@jiekang Sure thing, I've adjusted the indentation of the constructor.

File: src/app/dashboard-widgets/work-item-widget/work-item-widget.component.ts [hunk 1]
```diff
@@ -48,15 +49,37 @@ export class WorkItemWidgetComponent implements OnInit {
...
+    });
   }
 
-  ngOnInit() {
-    this.contextPath = this.contexts.current.map(context => context.path);
+  private resetWorkItemCounts(): void {
+    this.myWorkItemsCount = this.myWorkItemsOpen = this.myWorkItemsInProgress = this.myWorkItemsResolved = 0;
+  }
+
+  private updateWorkItems(): void {
```
Comment (joshuawilson) [inline src/app/dashboard-widgets/work-item-widget/work-item-widget.component.ts]: What do you think of making this not `private` so you can test it?

Comment (aptmac) [reply src/app/dashboard-widgets/work-item-widget/work-item-widget.component.ts]: I'll try and write a couple unit tests without changing access modifier

Comment (aptmac) [reply src/app/dashboard-widgets/work-item-widget/work-item-widget.component.ts]: Update: I've added unit tests that test the behaviour of the `updateWorkItems` method

**Comment by aptmac** [comment]
I've introduced two more commits, the first cleans up some of the code in the spec file, and the later adds unit tests that cover the behaviour of `updateWorkItems`

**Comment by joshuawilson** [comment]
please rebase

**Pull request merged**